### PR TITLE
Fix repository verification exceptions.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: lib/src/repository/check_repository.dart [hunk 1]
```diff
@@ -73,8 +74,34 @@ Future<Repository?> checkRepository(PackageContext context) async {
...
+        //       unrelated, but badly formatted pubspec.yaml in the repository.
+        // ignore: prefer_typing_uninitialized_variables
+        var yamlDoc;
+        try {
+          yamlDoc = yaml.loadYaml(content);
+        } on yaml.YamlException catch (e, st) {
+          log.info('Invalid yaml file: $path', e, st);
+        } on ArgumentError catch (e, st) {
+          log.info('Invalid yaml file: $path', e, st);
+        }
```
Comment (jonasfj) [inline lib/src/repository/check_repository.dart:91]: Don't you need to handle `FormatException` too?

File: lib/src/repository/check_repository.dart [hunk 2]
```diff
@@ -73,8 +74,34 @@ Future<Repository?> checkRepository(PackageContext context) async {
...
+        if (yamlDoc == null) {
+          return _PubspecMatch(path, false,
+              '`$path` from the repository is not a valid YAML document.');
+        }
+
+        late final Pubspec gitPubspec;
+        try {
+          gitPubspec = Pubspec.parseYaml(content);
+        } catch (e, st) {
```
Comment (jonasfj) [inline lib/src/repository/check_repository.dart:100]: Maybe only handle `FormatException` what can this really throw?

**Review by jonasfj — APPROVED** [review]

File: lib/src/repository/check_repository.dart (hunk 1 above)
Comment (isoos) [reply lib/src/repository/check_repository.dart:91]: Done.

File: lib/src/repository/check_repository.dart (hunk 2 above)
Comment (isoos) [reply lib/src/repository/check_repository.dart]: Done.

**Review by jonasfj — APPROVED** [review]

**Pull request merged**